sources/advertisements/VASTParser.swift

```diff
@@ -491,6 +504,21 @@ enum VASTParser {
         
         return delegateStack
     }
+    
+    static func parseTime(from string: String) -> Double? {
+        let components = string.components(separatedBy: ":")
+        guard components.count == 3 else { return nil }
+        return components
+            .compactMap(Double.init)
+            .enumerated()
+            .map {
+                let multiplier = pow(60, abs($0.offset - 2)) as NSNumber
```

Comment:
Why so complex? Please explain how this works.

Reply:
It is much shorter than if I write it as `components[0] * 3600` and so on.